src/Controls/tests/DeviceTests/Elements/TextInput/TextInputTests.cs

```diff
@@ -1,24 +1,338 @@
+		[ClassData(typeof(TextTransformCases))]
+		public async Task TextTransformUpdated(string text, TextTransform transform, string expected)
+		{
+			var control = new TView() { Text = text };
+			var handler = await CreateHandlerAsync<THandler>(control);
+			await InvokeOnMainThreadAsync(() => control.TextTransform = transform);
+			var platformText = await GetPlatformText(handler);
+			Assert.Equal(expected, platformText);
+		}
+
+		[Theory(DisplayName = "Unset CursorPosition is kept at zero at initialization with TextTransform")]
+		[InlineData("This is a test!!!")]
+		[InlineData("a")]
+		[InlineData("")]
+		[InlineData(" ")]
+		public async Task UnsetCursorPositionKeepsToZeroOnInitializationWithTextTransform(string text)
+		{
+			var editor = new TView
+			{
+				Text = text,
+				TextTransform = TextTransform.Uppercase
+			};
+
+			await ValidatePropertyInitValue<int, THandler>(
+				editor,
+				() => editor.CursorPosition,
+				GetPlatformCursorPosition,
+				0);
+		}
+
+		[Theory(DisplayName = "Unset SelectionLength is kept at zero at initialization with TextTransform")]
+		[InlineData("This is a test!!!")]
+		[InlineData("a")]
+		[InlineData("")]
+		[InlineData(" ")]
+		public async Task UnsetSelectionLengthKeepsToZeroOnInitializationWithTextTransform(string text)
+		{
+			var editor = new TView
+			{
+				Text = text,
+				TextTransform = TextTransform.Uppercase
+			};
+
+			await ValidatePropertyInitValue<int, THandler>(
+				editor,
+				() => editor.SelectionLength,
+				GetPlatformSelectionLength,
+				0);
+		}
+
+		[Theory(DisplayName = "CursorPosition Initializes Correctly")]
+		[InlineData(2)]
+		public async Task CursorPositionInitializesCorrectly(int initialPosition)
+		{
+			var entry = new TView
+			{
+				Text = "This is TEXT!",
+				CursorPosition = initialPosition
+			};
+
+			await ValidatePropertyInitValue<int, THandler>(
+				entry,
+				() => entry.CursorPosition,
+				GetPlatformCursorPosition,
+				initialPosition);
+		}
+
+		[Theory]
+		[InlineData(2)]
+		public async Task CursorPositionInitializesCorrectlyWithUpdateCursorPositionLast(int initialPosition)
+		{
+			var entry = new TView
+			{
+				Text = "This is TEXT!",
+				CursorPosition = initialPosition
+			};
+
+			var value = await GetValueAsync<int, THandler>(entry, handler =>
+			{
+				handler.UpdateValue(nameof(ITextInput.CursorPosition));
+				return GetPlatformCursorPosition(handler);
+			});
+
+			Assert.Equal(initialPosition, value);
+		}
+
+		[Theory]
+		[InlineData(2)]
+		public async Task CursorPositionInitializesCorrectlyWithUpdateTextLast(int initialPosition)
+		{
+			var entry = new TView
+			{
+				Text = "This is TEXT!",
+				CursorPosition = initialPosition
+			};
+
+			var value = await GetValueAsync<int, THandler>(entry, handler =>
+			{
+				handler.UpdateValue(nameof(ITextInput.Text));
+				return GetPlatformCursorPosition(handler);
+			});
+
+			Assert.Equal(initialPosition, value);
+		}
```

Comment:
These 2 tests cause failures depending on the order of properties in the mapper table. Currently if Text is first, then it is passing, but if it happens to have Text after the CursorPosition (which will be the case in my other PRs because I am fixing the mappers) then it fails.